pkg/auth/api/user/user_actions_test.go

```diff
@@ -0,0 +1,81 @@
+package user
+
+import (
+	"testing"
+)
+
+func TestValidatePassword(t *testing.T) {
+	tests := []struct {
+		name       string
+		username   string
+		password   string
+		expectsErr bool
+	}{
+		{
+			name:       "password too short",
+			username:   "admin",
+			password:   "admin",
+			expectsErr: true,
+		},
+		{
+			name:       "username equals password min length",
+			username:   "passwordpass",
+			password:   "passwordpass",
+			expectsErr: true,
+		},
+		{
+			name:       "password almost contains username begin",
```

Comment:
These whole "almost contains" should be replaced with a single "almost match" case.